### PR TITLE
Decompile unk_0202055C

### PR DESCRIPTION
**Pull request by abaresk**
First decompiled file :)

### CODE REVIEW

**Review by tgsm — CHANGES_REQUESTED** [review]

File: src/unk_0202055C.c [hunk 1]
```diff
@@ -0,0 +1,51 @@
+#include "global.h"
```
[the remark this answers is not in the document]
Comment (abaresk) [reply src/unk_0202055C.c]: Thanks for the tip!

File: src/unk_0202055C.c [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+void sub_0202055C(u32 a0, u32 a1) {
+    void *ptr;
+
+    GF_ASSERT(sUnknown == NULL);
+    ptr = AllocFromHeap(a1, sizeof(Unk0202055C));
+    sUnknown = ptr;
+    GF_ASSERT(ptr != NULL);
+
+    ptr = AllocFromHeap(a1, a0 << 4);
```
[the remark this answers is not in the document]
Comment (abaresk) [reply src/unk_0202055C.c]: Done

File: include/unk_0202055C.h [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#ifndef POKEHEARTGOLD_UNK_0202055C_H
+#define POKEHEARTGOLD_UNK_0202055C_H
+
+typedef struct {
+    u32 unk0;
+    u32 unk4;
+    void *unk8;
+} Unk0202055C;
+
```
[the remark this answers is not in the document]
Comment (abaresk) [reply include/unk_0202055C.h:9]: Added

File: src/unk_0202055C.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
+#include "heap.h"
+#include "unk_0202055C.h"
+
+static Unk0202055C *sUnknown;
+
+void sub_0202055C(u32 a0, HeapID a1) {
```
Comment (tgsm) [inline src/unk_0202055C.c:6]: a1 can be named `heap_id` (or `heapId` depending on your preference)

File: src/unk_0202055C.c [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    sUnknown->unk0 = a0;
+    sUnknown->unk4 = 0;
+    NNS_GfdInitVramTransferManager(sUnknown->unk8, sUnknown->unk0);
+}
+
+void sub_020205AC(void) {
+    GF_ASSERT(sUnknown != NULL);
+    FreeToHeap(sUnknown->unk8);
+    FreeToHeap(sUnknown);
+    sUnknown = 0;
```
Comment (tgsm) [inline src/unk_0202055C.c:26]:
```suggestion
    sUnknown = NULL;
```

File: src/unk_0202055C.c [hunk 6]
```diff
@@ -0,0 +1,51 @@
+#include "heap.h"
+#include "unk_0202055C.h"
+
+static Unk0202055C *sUnknown;
+
+void sub_0202055C(u32 a0, HeapID a1) {
+    void *ptr;
+
+    GF_ASSERT(sUnknown == NULL);
+    ptr = AllocFromHeap(a1, sizeof(Unk0202055C));
```
Comment (PikalaxALT) [inline src/unk_0202055C.c:10]: Is this intermediate var even necessary to match?

**Review by PikalaxALT — CHANGES_REQUESTED** [review]

File: src/unk_0202055C.c (hunk 6 above)
Comment (abaresk) [reply src/unk_0202055C.c]: Nope, removed

File: src/unk_0202055C.c (hunk 4 above)
Comment (abaresk) [reply src/unk_0202055C.c]: Done

File: src/unk_0202055C.c (hunk 5 above)
Comment (abaresk) [reply src/unk_0202055C.c]: Done

**Review by tgsm — APPROVED** [review]

**Pull request merged**